andela/ah-infinity-stones
# 161966913 password reset through email

#### What does this PR do?
- Implement password reset through email .
#### Description of Task to be completed?
- Implement sending password reset link to email.
- Implement password reset through email notification. 
- Implement tests
#### How should this be manually tested?
1. `git clone -b https://github.com/andela/ah-infinity-stones` ft-User-able-to-favorite-article-61966922  and `$ cd ah-infinity-stones`
2. Create and activate virtual env
- `virtualenv  -p python3 venv`
- `$ source venv/bin/activate`
3. `$ pip3 install -r requirements.txt` to install the dependencies
4. Setup environment variables. Copy all from below and paste them into the .env file. Update the values to your choice.
```export DATABASE_USER='ke-c17rn9b4fvh3'
export DATABASE_NAME='articles_db'
export DATABASE_PASSWORD='admin'
export DATABASE_TEST='articles_db_test'
export EMAIL_HOST_USER='domisemak@gmail.com'
export EMAIL_HOST_PASSWORD='Dommy2018'
export SECRET_KEY='7pgozr2jn7zs_o%i8id6=rddie!*0f0qy3$oy$(8231i^4*@u3''
```
Activate the env variables by `$source .env`.
5. Run migrations: `$ python3 manage.py migrate`
6. Run app - `$ python3 manage.py runserver`
7. Acess `http://127.0.0.1:8000/api/user/password-reset/
` through [Postman](https://www.getpostman.com/), and enter email to get a notification.
```
{ 
"email": "username@user.com"	
}
}
```
Check your email and click the link to reset password.  Use the link to test password reset.
8. Run tests` python3 manage.py test authors.apps.articles.tests.test_views.ArticleFavoriteTestCase`
###What are the relevant pivotal tracker stories?###
[[#161966922]](https://www.pivotaltracker.com/n/projects/2225605)

#### Screenshots
Passing tests:
<img width="1066" alt="screen shot 2018-12-17 at 06 26 10" src="https://user-images.githubusercontent.com/33648833/50064971-155b4980-01c5-11e9-8202-04cbe74c0065.png">

Password Reset email notification:
<img width="599" alt="screen shot 2018-12-17 at 06 29 43" src="https://user-images.githubusercontent.com/33648833/50065011-4b98c900-01c5-11e9-81c2-8160a2cf5a18.png">
